packages/compiler-cli/src/ngtsc/transform/src/declaration.ts

```diff
@@ -239,32 +239,49 @@ export class ReturnTypeTransform implements DtsTransform {
   }
 
   transformClassElement(element: ts.ClassElement, imports: ImportManager): ts.ClassElement {
-    if (!ts.isMethodSignature(element)) {
-      return element;
-    }
-
-    const original = ts.getOriginalNode(element) as ts.MethodDeclaration;
-    if (!this.typeReplacements.has(original)) {
-      return element;
+    // // TODO(alan-agius4): Remove when we no longer support TS 3.9
+    // TS < 3.9
+    if (ts.isMethodSignature(element)) {
+      const original = ts.getOriginalNode(element, ts.isMethodDeclaration);
```

Comment:
I suspect this is currently broken. This has to be:

```suggestion
      const original = ts.getOriginalNode(element, ts.isMethodSignature);
```

but I'm not sure if the resulting type of `ts.MethodSignature` would work, given that we previously casted it to `ts.MethodDeclaration` (and now I'm not even sure if that cast was right).